storage/client/queue/src/main/java/com/azure/storage/queue/QueueServiceAsyncClient.java

```diff
@@ -0,0 +1,342 @@
+     */
+    QueueServiceAsyncClient(URL endpoint, HttpPipeline httpPipeline) {
+        this.client = new AzureQueueStorageBuilder().pipeline(httpPipeline)
+            .url(endpoint.toString())
+            .build();
+    }
+
+    /**
+     * @return the URL of the storage queue
+     * @throws RuntimeException If the queue service is using a malformed URL.
+     */
+    public URL getQueueServiceUrl() {
+        try {
+            return new URL(client.url());
+        } catch (MalformedURLException ex) {
+            throw new RuntimeException("Storage account URL is malformed");
+        }
+    }
+
+    /**
+     * Constructs a QueueAsyncClient that interacts with the specified queue.
+     *
+     * This will not create the queue in the storage account if it doesn't exist.
+     *
+     * @param queueName Name of the queue
+     * @return QueueAsyncClient that interacts with the specified queue
+     */
+    public QueueAsyncClient getQueueAsyncClient(String queueName) {
+        return new QueueAsyncClient(client, queueName);
+    }
+
+    /**
+     * Creates a queue in the storage account with the specified name and returns a QueueAsyncClient to interact
+     * with it.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Create the queue "test"</p>
+     *
+     * {@codesnippet com.azure.storage.queue.queueServiceAsyncClient.createQueue#string}
+     *
+     * @param queueName Name of the queue
+     * @return A response containing the QueueAsyncClient and the status of creating the queue
+     * @throws StorageErrorException If a queue with the same name and different metadata already exists
+     */
+    public Mono<Response<QueueAsyncClient>> createQueue(String queueName) {
+        return createQueue(queueName, null);
+    }
+
+    /**
+     * Creates a queue in the storage account with the specified name and metadata and returns a QueueAsyncClient to
+     * interact with it.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Create the queue "test" with metadata "queue:metadata"</p>
+     *
+     * <pre>
+     * client.createQueue("test", Collections.singletonMap("queue", "metadata"))
+     *     .subscribe(response -&gt; System.out.printf("Creating the queue completed with status code %d", response.statusCode()));
+     * </pre>
+     *
+     * @param queueName Name of the queue
+     * @param metadata Metadata to associate with the queue
+     * @return A response containing the QueueAsyncClient and the status of creating the queue
+     * @throws StorageErrorException If a queue with the same name and different metadata already exists
+     */
+    public Mono<Response<QueueAsyncClient>> createQueue(String queueName, Map<String, String> metadata) {
+        QueueAsyncClient queueAsyncClient = new QueueAsyncClient(client, queueName);
+
+        return queueAsyncClient.create(metadata)
+            .map(response -> new SimpleResponse<>(response, queueAsyncClient));
```

Comment:
Blobs has a postResponseProcess helper method to convert the StorageErrorException into a StorageException, we should standardize this across the Storage libraries.

Reply:
Good call out. I can add a feature issue in epic. I's wish we push the basic one to feature branch first. Then address these things.

Reply:
#4390